Add sort by widget compatibility (#514)
meilisearch/instant-meilisearch: README.md

No compatibility because MeiliSearch does not support Rules.

### ✅ SortBy

[Sort by references](https://www.algolia.com/doc/api-reference/widgets/sort-by/js/)

> The `sortBy` widget is used to create multiple sort formulas. Allowing a user to change the way hits are sorted.

Reviewer: ```suggestion
The `sortBy` widget is used to create multiple sort formulas. Allowing a user to change the way hits are sorted at query time.
```

Author: The `sortBy` also proposes to only change the `index` without a sort `id:asc` for example. Which means these would not be sorted at query time

Reviewer: You can dismiss my proposal, the previous part is still true. The idea of "at query time" is to say that the user has some control and performs an action. 🤓

- ✅ container: The CSS Selector or HTMLElement to insert the widget into. _required_
- ✅ items: The list of different sorting possibilities. _required_
- ✅ cssClasses: The CSS classes to override.
- ✅ transformItems: function receiving the items, called before displaying them.

The usage of the `SortBy` widget differs from the one found in Algolia's documentation. In instant-meilisearch the following is possible:

- Sort using different indexes.
- Different `sort` rules on the same index.

The items list is composed of objects containing every sort possibilities you want to provide to your user. Each objects must contain two fields:
  - `label`: What is showcased on the user interface ex: `Sort by Ascending Price`
  - `value`: The sort formula.

#### Sort formula

> The sort formula is synthaxed like this: `index:attribute:order`.

Reviewer: ```suggestion
A sort formula is expressed like this: `index:attribute:order`.
```

`index` is mandatory, and when adding `attribute:order`, they must always be added together.

> When sorting on an attribute, the attribute has to be added in the [`SortableAttribute`](#link-to-doc) setting on your index.

Reviewer: ```suggestion
When sorting on an attribute, the attribute has to be added in the [`SortableAttributes`](#link-to-doc) settings on your index.
```

Author: Not sure about the `s` to settings! But I agree with the first correction

Author: ```suggestion
When sorting on an attribute, the attribute has to be added in the [`SortableAttributes`](#link-to-doc) setting on your index.
```

Example:
```js
[
  { label: 'Sort By Price', value: 'clothes:price:asc' }
]
```

In this scenario, in the `clothes` index, we want the price to be sorted in an ascending way. For this formula to be valid, `price` must be added in the `sortableAttributes` settings of the `clothes` index.

#### Relevancy

> The importance of the sort value is determined by the [`ranking-rules`](https://docs.meilisearch.com/learn/core_concepts/relevancy.html#ranking-rules) of each index. It is possible to change the ranking-rules in the index settings. Or make multiple indexes with different ranking-rules order. See [relevancy guide](https://docs.meilisearch.com/learn/core_concepts/relevancy.html#relevancy).

Reviewer: ```suggestion
The importance of the sort ranking rule is determined by its position among the [`ranking-rules](https://docs.meilisearch.com/learn/core_concepts/relevancy.html#ranking-rules) settings. It is possible to change the sort ranking rule position to favor exhaustivity over relevancy. See [relevancy guide](https://docs.meilisearch.com/learn/core_concepts/relevancy.html#relevancy).
```

Author: I see where you are going with this, can I suggest a counter suggestion?

```suggestion
The impact sorting has on the returned hits is determined by the [`ranking-rules`](https://docs.meilisearch.com/learn/core_concepts/relevancy.html#ranking-rules) ordered list of each index. The `sort` ranking-rule position in the list makes sorting documents more or less important than other rules. If you want to change the sort impact on the relevancy, it is possible to change it in the [ranking-rule setting](https://docs.meilisearch.com/reference/api/ranking_rules.html#update-ranking-rules). For example, to favor exhaustivity over relevancy.

See [relevancy guide](https://docs.meilisearch.com/learn/core_concepts/relevancy.html#relevancy).
```

Author: I see where you are going with this, can I suggest a counter suggestion?

```suggestion
The impact sorting has on the returned hits is determined by the [`ranking-rules`](https://docs.meilisearch.com/learn/core_concepts/relevancy.html#ranking-rules) ordered list of each index. The `sort` ranking-rule position in the list makes sorting documents more or less important than other rules. If you want to change the sort impact on the relevancy, it is possible to change it in the [ranking-rule setting](https://docs.meilisearch.com/reference/api/ranking_rules.html#update-ranking-rules). For example, to favor exhaustivity over relevancy.

See [relevancy guide](https://docs.meilisearch.com/learn/core_concepts/relevancy.html#relevancy).
```
